addons/website/static/src/snippets/s_share/000.js

```diff
@@ -75,10 +74,19 @@ const ShareWidget = publicWidget.Widget.extend({
                     return `${a + title}%20${url + c}`;
                 }
                 return a + title + c;
-            })
-            .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
             });
+        const urlObject = new URL(aEl.href);
+        if (urlObject.searchParams.has("media")) {
+            const ogImageEl = document.querySelector("meta[property='og:image']");
+            // Some pages (/profile/user/ID) don't have an image to share.
+            if (ogImageEl) {
+                const media = encodeURIComponent(ogImageEl.content);
+                urlObject.searchParams.set("media", media);
+            } else {
+                urlObject.searchParams.delete("media");
```

Comment:
@Guillaume-gdi We'll get the right implementation ! :smile: ... but I'll be annoying one last time :smile: 

Actually, since you delete the "media" part from the URL, I think this is actually not right still: what if the og:image changes in a custo so that the sharer uses an updated image on second click. Actually, I tested, and I am surprise we do not do that already for this flow to work:
- Go on the "Customizable Desk" product
- Click on the pinterest share button -> you get a preview of the standard customizable desk
- Close the sharer
- Switch the "black" color for the desk
- Click on the pinterest share button again -> the URL you are about to share is the one of the black customizable desk but the image is still the standard one.

This could actually qualify as a bug to me (strange to share the product.product (variant) URL but the product.template (main product) image). At least to be fixed in master IMO. Probably create a task for later so that you can focus on other things, we should also probably add the "media" part in other sharer links. Meanwhile, we cannot suppose some custo is not updating the og:image properly via custom JS so that the sharer uses it. Maybe do this:
- As before, keep the `media=whatever` part in the `href` of the `<a>` element.
- But remove that `media=` part for the `window.open` call just below

Reply:
> Variant image not updated for the sharer [...]

The OG image is never updated to the variant. I already reported it to @paju-odoo :+1: . If ecommerce don't fix it soon, I can do it after the Odoo 17 release. @paju-odoo do you have a task ID for the bug I reported to you on discord ? :arrow_down: 
![image](https://github.com/odoo/odoo/assets/78849981/34eec3a4-1a66-484e-9a84-539182693add)

> Leave media={XXX} in href

Indeed, good comment. It's done :heavy_check_mark:

Reply:
> > Variant image not updated for the sharer [...]
> 
> The OG image is never updated to the variant. I already reported it to @paju-odoo 👍 . If ecommerce don't fix it soon, I can do it after the Odoo 17 release. @paju-odoo do you have a task ID for the bug I reported to you on discord ? ⬇️ ![image](https://user-images.githubusercontent.com/78849981/270608471-34eec3a4-1a66-484e-9a84-539182693add.png)
> 
> > Leave media={XXX} in href
> 
> Indeed, good comment. It's done ✔️

Hi @Guillaume-gdi,
I created a feedback for that issue but it's not yet in our pipe, due to the great amount of work we have for 17... It's not a priority. Here is the task link, if you wish to assign yourself and work on it after the release :) 
https://www.odoo.com/web#id=3485494&cids=1&menu_id=4720&action=333&active_id=809&model=project.task&view_type=form